pkg/arvo/app/permission-hook.hoon

```diff
@@ -0,0 +1,282 @@
+      [~ this]
+    :_  this(synced (~(del by synced) path.diff))
+    :~  (permission-poke diff)
+        [ost.bol %pull [%permission path.diff] [src.bol %permission-hook] ~]
+    ==
+  ==
+::
+++  change-foreign-permission
+  |=  [pax=path diff=permission-update]
+  ^-  (quip move _this)
+  ?>  ?=([* ^] pax)
+  =/  owner-access=(unit owner-access)  (~(get by synced) pax)
+  :_  this
+  ?~  owner-access  ~
+  ?.  =(src.bol ship.u.owner-access)  ~
+  [(permission-poke diff)]~
+::
+++  quit-subscriptions
+  |=  [kind=?(%add %remove) pax=path who=(set ship) access-paths=(set path)]
+  ^-  (list move)
+  =/  perm  (permission-scry pax)
+  ?.  ?|
+        ?&(=(kind.perm %black) =(kind %add))
+        ?&(=(kind.perm %white) =(kind %remove))
+      ==
+    ::  if allow, do nothing
+    ~    
+  =/  sup
+    %-  ~(gas by *(map [ship path] bone))
+    %+  turn  ~(tap by sup.bol)
+    |=([=bone anchor=[ship path]] [anchor bone])
+  ::  if ban, iterate through
+  ::  all ships that have been banned
+  ::  and all affected paths that have had their permissions changed
+  ::  then quit their subscriptions
+  ::
+  %-  zing
+  %+  turn  ~(tap in who)
+  |=  check-ship=ship
+  ^-  (list move)
+  %-  zing
+  %+  turn  ~(tap in access-paths)
+  |=  access-path=path
+  ^-  (list move)
+  =/  bne  (~(get by sup) [check-ship [%permission access-path]])
+  ?~(bne ~ [u.bne %quit ~]~)
+::
+++  quit
+  |=  wir=wire
+  ^-  (quip move _this)
+  ~&  permission-hook-quit+wir
+  ?>  ?=([* ^] wir)
+  ?.  (~(has by synced) t.wir)
+    ::  no-op
+    [~ this]
+  =/  =owner-access  (~(got by synced) t.wir)
+  ~&  %permission-hook-resubscribe
+  :_  (track-bone wir)
+  [ost.bol %peer wir [ship.owner-access %permission-hook] wir]~
+::
+++  reap
+  |=  [wir=wire saw=(unit tang)]
+  ^-  (quip move _this)
+  ?~  saw
+    [~ this]
+  ?>  ?=(^ wir)
+  :_  this(synced (~(del by synced) t.wir))
+  %.  ~
+  %-  slog
+  :*  leaf+"permission-hook failed subscribe on {(spud t.wir)}"
+      leaf+"stack trace:"
+      u.saw
+  ==
+::
+++  permission-scry
+  |=  pax=path
+  ^-  permission
+  =.  pax  ;:(weld /=permission-store/(scot %da now.bol)/permission pax /noun)
+  (need .^((unit permission) %gx pax))
+::
+++  permitted-scry
+  |=  pax=path
+  ^-  ?
+  .^(? %gx ;:(weld /=permission-store/(scot %da now.bol)/permitted pax /noun))
+::
+++  permission-poke
+  |=  act=permission-action
+  ^-  move
+  [ost.bol %poke / [our.bol %permission-store] [%permission-action act]]
+::
+++  update-subscribers
+  |=  [pax=path upd=permission-update]
+  ^-  (list move)
+  %+  turn  (prey:pubsub:userlib pax bol)
+  |=  [=bone *]
+  [bone %diff %permission-update upd]
+::
+++  track-bone
+  |=  wir=wire
+  ^+  this
+  =/  bnd  (~(get by boned) wir)
+  ?^  bnd
+    this(boned (~(put by boned) wir (snoc u.bnd ost.bol)))
+  this(boned (~(put by boned) wir [ost.bol]~))
+::
+++  pull-wire
+  |=  pax=path
+  ^-  (list move)
+  ?>  ?=([* ^] pax)
+  =/  bnd  (~(get by boned) pax)
+  ?~  bnd  ~
```

Comment:
Similarly, you could consider using `(~(get ja boned) pax)` here. (Or rather, `%+  turn  (~(get ja boned) pax)` below.)